src/offerings.ts

```diff
@@ -325,3 +380,181 @@ export enum PRORATION_MODE {
      */
     IMMEDIATE_AND_CHARGE_FULL_PRICE = 5,
 }
+
+/**
+ * Contains all details associated with a SubscriptionOption
+ * Used only for Google
+ */
+export interface SubscriptionOption {
+    /**
+     * Identifier of the subscription option
+     * If this SubscriptionOption represents a base plan, this will be the basePlanId.
+     * If it represents an offer, it will be {basePlanId}:{offerId}
+     */
+    readonly id: string;
+
+    /**
+     * Identifier of the StoreProduct associated with this SubscriptionOption
+     * This will be {subId}:{basePlanId}
+     */
+    readonly storeProductId: string;
+
+    /**
+     * Identifer of the subscription associated with this SubscriptionOption
+     * This will be {subId}
+     */
+    readonly productId: string;
+
+    /**
+     * Pricing phases defining a user's payment plan for the product over time.
+     */
+    readonly pricingPhases: PricingPhase[];
+
+    /**
+     * Tags defined on the base plan or offer. Empty for Amazon.
+     */
+    readonly tags: string[];
+
+    /**
+     * True if this SubscriptionOption represents a subscription base plan (rather than an offer).
+     */
+    readonly isBasePlan: boolean;
+
+    /**
+     * The subscription period of fullPricePhase (after free and intro trials).
+     */
+    readonly billingPeriod: Period | null;
+
+    /**
+     * The full price PricingPhase of the subscription.
+     * Looks for the last price phase of the SubscriptionOption.
+     */
+    readonly fullPricePhase: PricingPhase | null;
+
+    /**
+     * The free trial PricingPhase of the subscription.
+     * Looks for the first pricing phase of the SubscriptionOption where amountMicros is 0.
+     * There can be a freeTrialPhase and an introductoryPhase in the same SubscriptionOption.
+     */
+    readonly freePhase: PricingPhase | null;
+
+    /**
+     * The intro trial PricingPhase of the subscription.
+     * Looks for the first pricing phase of the SubscriptionOption where amountMicros is greater than 0.
+     * There can be a freeTrialPhase and an introductoryPhase in the same SubscriptionOption.
+     */
+    readonly introPhase: PricingPhase | null;
+
+    /**
+     * Offering identifier the subscription option was presented from
+     */
+    readonly presentedOfferingIdentifier: string | null;
+}
+
+/**
+ * Contains all the details associated with a PricingPhase
+ */
+export interface PricingPhase {
+    /**
+     * Billing period for which the PricingPhase applies
+     */
+    readonly billingPeriod: Period;
+
+    /**
+     * Recurrence mode of the PricingPhase
+     */
+    readonly recurrenceMode: RECURRENCE_MODE | null;
```

Comment:
Can this be nullable? It's not in Android

Reply:
Looks like it's using the identifier property from android, which can be null if `unknown`: https://github.com/RevenueCat/purchases-android/blob/main/public/src/main/java/com/revenuecat/purchases/models/RecurrenceMode.kt#L19

Reply:
I see! Makes sense then!